test/TestExtensions/API.Test/VSSolutionHelper.cs

```diff
@@ -467,7 +467,7 @@ public static async Task<EnvDTE.Project> GetProjectAsync(EnvDTE80.Solution2 solu
             return solution2
                 .Projects
                 .Cast<EnvDTE.Project>()
-                .FirstOrDefault(project => project.Name.Equals(projectName, StringComparison.OrdinalIgnoreCase));
+                .FirstOrDefault(project => { ThreadHelper.ThrowIfNotOnUIThread(); return project.Name.Equals(projectName, StringComparison.OrdinalIgnoreCase); });
```

Comment:
I think `ThreadHelper.ThrowIfNotOnUIThread();` check is redundant because the caller is on the main thread unless I am missing something here

Reply:
This is a false positive by the analyzer. 

See the linked issue: https://github.com/microsoft/vs-threading/issues/577